### PR TITLE
Fix https://github.com/jasp-stats/jasp-test-release/issues/289

### PR DESCRIPTION
**Pull request by koenderks**
Add an error in the decision matrix when there is only 1 numeric variable remaining after removing all factors from the variables.

### CODE REVIEW

File: JASP-Engine/JASP/R/commonMachineLearningClassification.R [hunk 1]
```diff
@@ -334,6 +334,11 @@
   variables <- variables[ !sapply(dataset[, .v(variables)], is.factor) ] # remove factors from boundary plot
   l <- length(variables)
 
+  if(l == 1){
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/commonMachineLearningClassification.R]: can `l` be `0` at this point, too?

Comment (koenderks) [reply JASP-Engine/JASP/R/commonMachineLearningClassification.R]: Yes, will fix by replacing with <= 1

**Comment by koenderks** [comment]
Now also changes some input variable types for analyses and fixes error in boosting deviance plot

**Comment by koenderks** [comment]
And adds data files for tims blogpost and jasp file for mine

**Pull request merged**